src/publishMain.ml

```diff
@@ -51,6 +51,27 @@ let tmp_source tmpdir url =
                      (Digest.string (OpamUrl.to_string url))
                      (OpamUrl.basename url))
 
+let upgrade_to_2_0 ?(local=true) opam0 =
+  OpamStd.Option.iter (fun opam ->
+      let opam2 = OpamFormatUpgrade.opam_file ~filename:opam0 opam in
+      if not (OpamFile.OPAM.equal opam2 opam) then
+        (OpamConsole.warning "%s has %s format, \
+                              which is not accepted on 2.0 repository.%s"
+           (if local then
+              "Opam file " ^ OpamFilename.to_string (OpamFile.filename opam0)
+            else "Downloaded opam file")
+           (OpamConsole.colorise `underline
+              (OpamVersion.to_string (OpamFile.OPAM.opam_version opam)))
+           (if not local then
+              OpamConsole.colorise `bold " Updating it."
+            else "");
+         if local &&
+            not (OpamConsole.confirm " Update it inplace to %s format?"
+                   (OpamConsole.colorise `bold "2.0"))
+         then OpamStd.Sys.exit_because `Aborted;
```

Comment:
We could actually continue here, just submitting the rewritten file and leaving the source as is. But it may be better this way.

Reply:
Like that, users can write, translate, and check themselves their opam 2.0 files instead of keeping 1.2 ones indefinitely.